Sources/XCHammer/XcodeTarget.swift

```diff
@@ -553,9 +553,21 @@ public class XcodeTarget: Hashable, Equatable {
     private static let librarySourceTypes = Set(["swift", "m", "mm", "cpp", "c", "cc", "S"])
 
     // Returns transitive linkable
-    lazy var transitiveDeps: Set<ProjectSpec.Dependency> = {
+    lazy var xcLinkableTransitiveDeps: Set<ProjectSpec.Dependency> = {
+        // By traversing the dependency graph through `transitiveTargets` we
+        // pick up many dependencies, that may or may not be on the linker
+        // command line. Consider other ways to do this. For now, reject known
+        // non legitimate dependency propagators.
+        let linkablePredicate: TraversalTransitionPredicate<XcodeTarget> = TraversalTransitionPredicate {
+            xcodeTarget -> Transition in
+            guard xcodeTarget.type != "_headermap" else {
+                return Transition.stop
```

Comment:
Would it be better to just have a tag on rules that shouldn’t traverse? I doubt this will be the last time we need to patch this logic as we add new rules

Reply:
The linker deps should now be extractable from bazel, this should just use that. Otherwise, it isn't maintainable to re-implement bazel's/rule logic here.